ansible_base/resource_registry/models/resource.py

```diff
@@ -95,7 +96,9 @@ def delete_resource(self):
             self.delete()
 
     @classmethod
-    def create_resource(cls, resource_type: ResourceType, resource_data: dict, ansible_id: str = None, service_id: str = None):
+    def create_resource(
+        cls, resource_type: ResourceType, resource_data: dict, ansible_id: Union[str, uuid.UUID, None] = None, service_id: Union[str, uuid.UUID, None] = None
```

Comment:
If we were targetting Python 3.10+, then we could rewrite the unions as `Optional[str | uuid.UUID]` which IMO is more readable.

But we're not, so we can't. :cry:

Reply:
I'm completely sure this will get better with time. If we have this check enabled, then the developer writing this is likely to re-consider passing in a `uuid.UUID` value simply for the point of simplifying the type hint. And that would be a _good_ thing! But my concern here is very strictly that type hints are _trustworthy_ in DAB to start with. I have wanted to roll out much more type hints in RBAC, as a part of better overall internal documentation, but hesitated because we didn't have this, and lately because it wasn't working.